solutions/usaco-647.mdx

```diff
@@ -2,22 +2,68 @@
 id: usaco-647
 source: USACO Gold 2016 Open
 title: '248'
-author: Melody Yu (Video), Qi Wang
+author: Melody Yu, Qi Wang
 ---
 
-[Official Analysis](http://www.usaco.org/current/data/sol_248_gold_open16.html)
+[Official Analysis (Java)](http://www.usaco.org/current/data/sol_248_gold_open16.html)
 
 ## Video Solution
 
 Note: The video solution might not be the same as other solutions. Code in C++.
 <Youtube id="OdHxOYICx-o" />
 
-
 ## Implementation
 
 **Time Complexity:** $\mathcal{O}(N^3)$
 
 <LanguageSection>
+<CPPSection>
+	
+```cpp
+#include <bits/stdc++.h>
+using namespace std;
+
+int main() {
+	freopen("248.in", "r", stdin);
+
+	int n;
+	cin >> n;
+	vector<int> a(n);
+
+	int bst = 0;
```

Comment:
ryan i'm begging you stop removing vowels from variable names

Reply:
ITS TOO LATE IM ADDICTED